polling_stations/apps/data_importers/management/commands/run_new_imports.py

```diff
@@ -0,0 +1,167 @@
+import shutil
+from pathlib import Path
+
+import boto3
+from django.apps import apps
+from django.core.management import BaseCommand, call_command
+
+import re
+import subprocess
+import sys
+
+
+def get_paths_changed(from_sha, to_sha):
+    args = ["git", "diff", "--name-only", from_sha, to_sha]
+    output = subprocess.check_output(args)
+    return output.decode().splitlines()
+
+
+def git_rev_parse(rev):
+    return subprocess.check_output(["git", "rev-parse", f"{rev}"]).decode().strip()
+
+
+def get_changed_scripts(changed):
+    if any_import_scripts(changed):
+        return [p for p in changed if is_import_script(p)]
+    return []
+
+
+def is_import_script(path):
+    if re.search(
+        r"polling_stations/apps/data_importers/management/commands/import_[^eoni].+\.py",
+        path,
+    ):
+        return True
+    return False
+
+
+def any_import_scripts(changed):
+    return any(is_import_script(path) for path in changed)
+
+
+def any_non_import_scripts(changed):
+    return any(not is_import_script(path) for path in changed)
+
+
+def stop_job():
+    sys.stdout.write("Using CircleCI Agent to stop jobs")
+    args = ["circleci-agent", "step", "halt"]
+    subprocess.check_output(args)
+
+
+def get_last_import_sha_from_ssm():
+    ssm_client = boto3.client("ssm")
+    response = ssm_client.get_parameter(Name="LAST_IMPORT_SHA")
+    return response["Parameter"]["Value"]
+
+
+class Command(BaseCommand):
+    help = """
+        Checks the files changed between two GIT commits
+        and decides if import scripts should be run
+    """
+
+    # Turn off auto system check for all apps
+    # We will manually run system checks only for the
+    # 'data_importers' and 'pollingstations' apps
+    requires_system_checks = []
+
+    summary = []
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "-f",
+            "--from-sha",
+            help="<Optional> The earliest commit hash against which to search for changed scripts. "
+            "Defaults to value of LAST_IMPORT_SHA in ssm parameter store",
+        )
+        parser.add_argument(
+            "-t",
+            "--to-sha",
+            help="<Optional> The latest commit hash against which to search for changed scripts. Defaults to HEAD",
+            default=git_rev_parse("HEAD"),
+        )
+        parser.add_argument("--post-deploy", action="store_true", default=False)
+
+    def run_scripts(self, changed_paths, opts):
+        for script in changed_paths:
+            script_path = Path(script)
+            try:
+                call_command(script_path.stem, **opts)
+                self.summary.append(
+                    (
+                        "INFO",
+                        f"Ran import script: {script_path.name}",
+                    )
+                )
+            except:
+                # usually we want to handle a specific exception, but in this situation
```

Comment:
Is this true? Why don't you want to fail CI if the import script can't run?

Reply:
Also, it's a good idea to at least catch `Exception`. This means that other

Reply:
Right, so my thinking here was that we don't want failed import scripts to block deploys. But I think I actually agree with you. Most of our import script changes are single commits, so it's simple enough to just drop that commit if there's an issue. I think I might go with your recommendation of catching `Exception` as a starting point, then refactor further to take it out completely.